src/main/java/org/folio/services/exchange/FinanceApiExchangeRateProvider.java

```diff
@@ -40,20 +36,25 @@ public ProviderContext getContext() {
 
   @Override
   public ExchangeRate getExchangeRate(ConversionQuery conversionQuery) {
-    return getExchangeRateFromService(conversionQuery)
-      .thenApply(exchangeRate -> {
-        ExchangeRateBuilder builder = new ExchangeRateBuilder(ConversionContext.of());
-        builder.setBase(conversionQuery.getBaseCurrency());
-        builder.setTerm(conversionQuery.getCurrency());
-        builder.setFactor(DefaultNumberValue.of(exchangeRate.getExchangeRate()));
-        return builder.build();
-      }).join();
+    var exchangeRate = getExchangeRateFromService(conversionQuery);
+
+    ExchangeRateBuilder builder = new ExchangeRateBuilder(ConversionContext.of());
+    builder.setBase(conversionQuery.getBaseCurrency());
+    builder.setTerm(conversionQuery.getCurrency());
+    builder.setFactor(DefaultNumberValue.of(exchangeRate.getExchangeRate()));
+    return builder.build();
   }
 
-  private CompletableFuture<org.folio.rest.acq.model.finance.ExchangeRate> getExchangeRateFromService(ConversionQuery conversionQuery) {
-    return CompletableFuture.supplyAsync(() -> financeExchangeRateService
-      .getExchangeRate(conversionQuery.getBaseCurrency().getCurrencyCode(), conversionQuery.getCurrency().getCurrencyCode(),
-        requestContext).join());
+  private org.folio.rest.acq.model.finance.ExchangeRate getExchangeRateFromService(ConversionQuery conversionQuery) {
+    double exchangeRate = MonetaryConversions.getExchangeRateProvider(IDENTITY, ECB)
+      .getExchangeRate(conversionQuery.getBaseCurrency(), conversionQuery.getCurrency())
```

Comment:
Is .getExchangeRate() blocking operation? if so we need to wrap into executeBlocking construction

Reply:
No need to do it here. It is already done in the place where this method used